tooling/reproducible/macos_repro_build_compare.sh

```diff
@@ -0,0 +1,673 @@
+#!/bin/bash
+# ********************************************************************************
+# Copyright (c) 2024 Contributors to the Eclipse Foundation
+#
+# See the NOTICE file(s) with this work for additional
+# information regarding copyright ownership.
+#
+# This program and the accompanying materials are made
+# available under the terms of the Apache Software License 2.0
+# which is available at https://www.apache.org/licenses/LICENSE-2.0.
+#
+# SPDX-License-Identifier: Apache-2.0
+# ********************************************************************************
+
+# This script examines the given sbom metadata file, and then builds the exact same binary
+# and then compares with the temurin jdk for the same build version, or the optionally supplied tarball_url.
+# Usage Notes:
+# 1. For MacOS, jq must be installed, and the architecture to be built must match the system this script is being executed on.
+# 2. This script will only work with xcode, and the executing user must have sudo permissions to run xcode-select -s
+# 3. This script requires that the correct versions of the sdk are installed and in the loaction defined in the MAC_SDK_LOCATION below.
+
+set -e
+
+# Check All 3 Params Are Supplied
+if [ "$#" -lt 2 ]; then
+  echo "Usage: $0 SBOM_URL/SBOM_PATH JDKZIP_URL/JDKZIP_PATH"
+  echo ""
+  echo "1. SBOM_URL/SBOM_PATH - should be the FULL path OR a URL to a Temurin JDK SBOM JSON file in CycloneDX Format"
+  echo "    eg. https://github.com/adoptium/temurin21-binaries/releases/download/jdk-21.0.3%2B9/OpenJDK21U-sbom_x64_mac_hotspot_21.0.3_9.json"
+  echo ""
+  echo "2. JDKZIP_URL/JDKZIP_PATH - should be the FULL path OR a URL to a Temurin Windows JDK Zip file"
+  echo "    eg. https://github.com/adoptium/temurin21-binaries/releases/download/jdk-21.0.3%2B9/OpenJDK21U-jdk_x64_mac_hotspot_21.0.3_9.tar.gz"
+  echo ""
+  exit 1
+fi
+
+# Read Parameters
+SBOM_URL="$1"
+TARBALL_URL="$2"
+
+# Constants Required By This Script
+# These Values Should Be Updated To Reflect The Build Environment
+# The Defaults Below Are Suitable For An Adoptium Windows Build Environment
+# Which Has Been Created Via The Ansible Infrastructure Playbooks
+
+WORK_DIR=/Users/jenkins/sfryer/comp-jdk-build
```

Comment:
I suspect WORK_DIR needs to be a argument or sub-folder?

Reply:
Fixed.